src/main/java/org/sigmah/client/ui/presenter/zone/OfflineBannerPresenter.java

```diff
@@ -22,565 +22,732 @@
+		view.getTraceHandle().addDomHandler(new MouseOutHandler() {
+
+			@Override
+			public void onMouseOut(MouseOutEvent event) {
+				traceLinkHover = false;
+				traceForceOpen = false;
+				updateTraceMenuVisibility();
+			}
+			
+		}, MouseOutEvent.getType());
+		
+        final OfflineMenuPanel menuPanel = view.getMenuPanel();
+		
+		menuPanel.addDomHandler(new MouseOverHandler() {
+			@Override
+			public void onMouseOver(MouseOverEvent event) {
+				menuHover = true;
+				updateMenuVisibility();
+			}			
+		}, MouseOverEvent.getType());
+		
+		menuPanel.addDomHandler(new MouseOutHandler() {
+			@Override
+			public void onMouseOut(MouseOutEvent event) {
+				menuHover = false;
+				forceOpen = false;
+				updateMenuVisibility();
+			}			
+		}, MouseOutEvent.getType());
+		
+		final TraceMenuPanel traceMenuPanel = view.getTraceMenuPanel();		
+		traceMenuPanel.addDomHandler(new MouseOverHandler() {
+			@Override
+			public void onMouseOver(MouseOverEvent event) {
+				traceMenuHover = true;
+				updateTraceMenuVisibility();
+			}
+			
+		}, MouseOverEvent.getType());
+		
+		traceMenuPanel.addDomHandler(new MouseOutHandler() {
+
+			@Override
+			public void onMouseOut(MouseOutEvent event) {
+				traceMenuHover = false;
+				traceForceOpen = false;
+				updateTraceMenuVisibility();
+			}
+			
+		}, MouseOutEvent.getType());
+		
+		
+        menuPanel.setSigmahUpdateDate(ApplicationCacheManager.getUpdateDate());
+        menuPanel.setDatabaseUpdateDate(getDatabaseUpdateDate());
+        
+        // Destroy local database
+        menuPanel.getRemoveOfflineDataAnchor().addClickHandler(new ClickHandler() {
+            @Override
+            public void onClick(ClickEvent event) {
+                confirmUserDatabaseRemoval();
+            }
+        });
+		
+        updateProgressBars();
+		
+		// Actions
+		
+		menuPanel.getUpdateDatabaseAnchor().addClickHandler(new ClickHandler() {
+			@Override
+			public void onClick(ClickEvent event) {
+				if (!view.isEnabled(menuPanel.getUpdateDatabaseAnchor())) {
+					return;
+				}
+				if (lastState == ApplicationState.READY_TO_SYNCHRONIZE) {
+					setMenuVisible(false);
+					pushAndPull();
+				} else if(lastState == ApplicationState.ONLINE) {
+					setMenuVisible(false);
+					pull();
+				}
+			}
+		});
+		
+		menuPanel.getTransferFilesAnchor().addClickHandler(new ClickHandler() {
+			@Override
+			public void onClick(ClickEvent event) {
+				if (!view.isEnabled(menuPanel.getTransferFilesAnchor())) {
+					return;
+				}
+				if (lastState == ApplicationState.ONLINE) {
+					setMenuVisible(false);
+					eventBus.navigate(Page.OFFLINE_SELECT_FILES);
+				}
+			}
+		});
+		
+		traceMenuPanel.getSendReportAnchor().addClickHandler(new ClickHandler() {
+			@Override
+			public void onClick(ClickEvent event) {
+				
+				executionAsyncDAO.getAllExecutions( new AsyncCallback<List<Execution>>() {
+						@Override
+						public void onFailure(Throwable caught) {
+							Log.error("Exception occurred ");
```

Comment:
If there is an error, the exception `caught` should be logged (passed as last argument of Log#error). If it is already logged elsewhere (I think so), that's fine but this log should then be `trace` or `debug` I guess.